### PR TITLE
add the fine-tuning results

### PR DESCRIPTION
**Pull request by RezaYazdaniAminabadi**

### CODE REVIEW

File: docs/_posts/2020-05-28-fastest-bert-training.md [hunk 1]
```diff
@@ -92,6 +92,32 @@ throughput by combining our software optimizations with the new hardware. We
...
+For the 16-GB V100, we can achieve up to 1.5x speedup while supporting 2x larger batch size per GPU. 
+On the other hand, we can support as large as 32 batch size (2.6x more than Pytorch) using 32GB of memory, while providing 1.3x speedup for the end-to-end fine-tune training. Note, that we use the best 
+samples-per-second to compute speedup for the cases that PyTorch runs OOM.
+
+| Micro Batch size | PyTorch | DeepSpeed | Speedup (x) |
+| ---------------- | ------- | --------- | ----------- |
+| 4                | 36.34   | 50.76     | 1.4         |
+| 6                | OOM     | 54.28     | 1.5         |
+| 8                | OOM     | 54.16     | 1.5         |
+Table 2. Samples/second for running Squad Fine-Tuning on NVIDIA V100 (16-GB) using PyTorch and DeepSpeed transformer kernels.
```
Comment (samyam) [inline docs/_posts/2020-05-28-fastest-bert-training.md:110]: The table captions are showing up inside the table. Can we put the table captions outside?

File: docs/_tutorials/bert-finetuning.md [hunk 2]
```diff
@@ -183,18 +184,46 @@ Note:
 1. `batch_size` is the maximum bath size of input data, all fine-tuning training data or prediction data shouldn't exceed this threshold, otherwise it will throw an exception. In the DeepSpeed configuration file micro batch size is defined as `train_micro_batch_size_per_gpu`, e.g. if it is set as 8 and prediction uses batch size of 12, we can use 12 as transformer kernel batch size, or using "--predict_batch_size" argument to set prediction batch size to 8 or a smaller number.
 2. `local_rank` in DeepSpeedTransformerConfig is used to assign the transformer kernel to the correct device. Since the model already runs set_device() before here, so does not need to be set here.
 
-For more details about the transformer kernel, please see [DeepSpeed Transformer Kernel](/transformer_kernel/) and [DeepSpeed Fast-Bert Training](/fast_bert/).
+For more details about the transformer kernel, please see [DeepSpeed Transformer Kernel](/_tutorials/transformer_kernel/) and [DeepSpeed Fast-Bert Training](/_posts/2020-05-28-fastest-bert-training/).
+
+### Increasing Micro-Batch Size
```
Comment (samyam) [inline docs/_tutorials/bert-finetuning.md:189]: Can we just call this section Tuning Micro-Batch Size?

File: docs/_posts/2020-05-28-fastest-bert-training.md (hunk 1 above)
Comment (RezaYazdaniAminabadi) [reply docs/_posts/2020-05-28-fastest-bert-training.md:110]: Sure, I will do that. Thanks

File: docs/_tutorials/bert-finetuning.md (hunk 2 above)
Comment (RezaYazdaniAminabadi) [reply docs/_tutorials/bert-finetuning.md:189]: Sure!

File: docs/_tutorials/bert-finetuning.md [hunk 3]
```diff
@@ -99,94 +104,121 @@ In the Baseline script you need to handle the all-reduce operation at the gradie
...
-2. `local_rank` in DeepSpeedTransformerConfig is used to assign the transformer kernel to the correct device. Since the model already runs set_device() before here, so does not need to be set here.
-
-For more details about the transformer kernel, please see our [usage tutorial](/tutorials/transformer_kernel/) and [technical deep dive](https://www.deepspeed.ai/news/2020/05/27/fastest-bert-training.html) on the fastest BERT training.
+Note: `batch_size` is the maximum bath size of input data, all fine-tuning
+training data or prediction data shouldn't exceed this threshold, otherwise it
+will throw an exception. In the DeepSpeed configuration file micro batch size
+is defined as `train_micro_batch_size_per_gpu`, e.g., if it is set as 8 then
+the `--predict_batch_size` should also be 8.
```
Comment (eltonzheng) [inline docs/_tutorials/bert-finetuning.md:199]: --predict_batch_size should be <= 8, or not exceeding 8.

File: docs/_tutorials/bert-finetuning.md [hunk 4]
```diff
@@ -99,94 +104,121 @@ In the Baseline script you need to handle the all-reduce operation at the gradie
...
+were 3e-5; We set the seeds to 9041 and 19068 for HuggingFace and TensorFlow
+models, respectively. The checkpoints used for each case are linked in the
+table below.
 
 | Case        | Model                                 | Precision | EM    | F1    |
 | ----------- | ------------------------------------- | --------- | ----- | ----- |
-| TensorFlow  | Bert-large-uncased-L-24_H-1024_A-16   | FP16      | 84.13 | 91.03 |
-| HuggingFace | Bert-large-uncased-whole-word-masking | FP16      | 87.27 | 93.33 |
+| TensorFlow  | [Bert-large-uncased-L-24_H-1024_A-16](https://storage.googleapis.com/bert_models/2018_10_18/uncased_L-24_H-1024_A-16.zip)   | FP16      | 84.13 | 91.03 |
+| HuggingFace | [Bert-large-uncased-whole-word-masking](https://s3.amazonaws.com/models.huggingface.co/bert/bert-large-uncased-whole-word-masking-pytorch_model.bin) | FP16      | 87.27 | 93.33 |
```
Comment (eltonzheng) [inline docs/_tutorials/bert-finetuning.md:137]: The two model links are same with what I downloaded & tested.

File: docs/_tutorials/bert-finetuning.md [hunk 5]
```diff
@@ -259,19 +298,17 @@ Table 4. The setting of memory-optimization flags for a range of micro-batch siz
 
 ### FineTuning model pre-trained with DeepSpeed Transformer Kernels
 
-Fine-tuning the model pre-trained using DeepSpeed Transformer and the recipe in [DeepSpeed Fast-Bert Training](/fast_bert/) should yield F1 score of 90.5 and is expected to increase if you let the pre-training longer than suggested in the tutorial. 
+Fine-tuning the model pre-trained using DeepSpeed Transformer and the recipe in [DeepSpeed Fast-Bert Training](/fast_bert/) should yield F1 score of 90.5 and is expected to increase if you let the pre-training longer than suggested in the tutorial.
```
Comment (ShadenSmith) [inline docs/_tutorials/bert-finetuning.md:304]: broken link: `/fast_bert/` should be `https://www.deepspeed.ai/news/2020/05/27/fastest-bert-training.html`

**Pull request merged**